target/linux/kirkwood/patches-6.6/300-nvmem.patch

```diff
@@ -0,0 +1,107 @@
+From 5ed34e7090dfe69ebac6366efb2dbc3006f42831 Mon Sep 17 00:00:00 2001
+From: Rosen Penev <rosenp@gmail.com>
+Date: Tue, 17 Sep 2024 16:59:28 -0700
+Subject: [PATCH] arch: arm: kirkwood: don't override mac address
+
+nvmem is now available to do so. Get rid of the now deprecated
+local-mac-address hack.
+
+of_get_ethdev_address works now.
+
+Signed-off-by: Rosen Penev <rosenp@gmail.com>
+---
+ arch/arm/mach-mvebu/kirkwood.c | 76 ----------------------------------
+ 1 file changed, 76 deletions(-)
+
+--- a/arch/arm/mach-mvebu/kirkwood.c
++++ b/arch/arm/mach-mvebu/kirkwood.c
+@@ -68,81 +68,6 @@ static void __init kirkwood_cpuidle_init
+ #define MV643XX_ETH_MAC_ADDR_LOW	0x0414
+ #define MV643XX_ETH_MAC_ADDR_HIGH	0x0418
+ 
+-static void __init kirkwood_dt_eth_fixup(void)
+-{
+-	struct device_node *np;
+-
+-	/*
+-	 * The ethernet interfaces forget the MAC address assigned by u-boot
+-	 * if the clocks are turned off. Usually, u-boot on kirkwood boards
+-	 * has no DT support to properly set local-mac-address property.
+-	 * As a workaround, we get the MAC address from mv643xx_eth registers
+-	 * and update the port device node if no valid MAC address is set.
+-	 */
+-	for_each_compatible_node(np, NULL, "marvell,kirkwood-eth-port") {
+-		struct device_node *pnp = of_get_parent(np);
+-		struct clk *clk;
+-		struct property *pmac;
+-		u8 tmpmac[ETH_ALEN];
+-		void __iomem *io;
+-		u8 *macaddr;
+-		u32 reg;
+-
+-		if (!pnp)
+-			continue;
+-
+-		/* skip disabled nodes or nodes with valid MAC address*/
+-		if (!of_device_is_available(pnp) ||
+-		    !of_get_mac_address(np, tmpmac))
```

Comment:
I think removing this will not be accapted upstream.

You could try to check if `of_get_mac_address()` returns `-EPROBE_DEFER` and then go to `eth_fixup_skip` here. In such a case it should get a correct mac address later.

Reply:
This is in early init. No way they would accept that.

Reply:
@hauke anyway, made the change. I'll try upstreaming but I'm skeptical.

Reply:
Oh wait this is completely wrong. You're confusing of_get_mac_address with of_get_etherdev_address.